src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machine = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machine.addAll(new BulldozerProducerImpl().get());
+        } else if (type == Excavator.class) {
+            machine.addAll(new ExcavatorProducerImpl().get());
+        } else if (type == Truck.class) {
+            machine.addAll(new TruckProducerImpl().get());
+        }
+        return machine;
```

Comment:
Consider creating a local variable for the machines list inside the `getAll` method to store the result from the producer and then return a new list created from that variable. This can prevent exposing the internal list structure and improve encapsulation.